docs/contrib/Logger.rst

```diff
@@ -1,29 +1,94 @@
+These are supported by automatic calls to the logger which will then prefix the appropriate severity type to the message. 
 
-Message Types
--------------
+Messaging Guidelines
+--------------------
 
+In addition to the proper use of message types, follow the guidelines below to compose messages for clarity, consistency and other guidelines:
+
+a. Grammar
+   i. Start with a capital letter and end with period besides well-known exceptions. Use capital letters for file formats and tool names. e.g., 
+      LEF, DEF, SPICE.
+   ii. After the first word’s capitalization, do not use capital letters (aside from obvious exceptions - RSMT, hCut, etc.
+   iii. Do not use exclamations. Severity must be communicated by message severity and clear implied or explicit action.
+   iv. Avoid long, verbose messages. Use commas to list and separate clauses in messages.
+   v. Spellcheck all messages using American English spellings.
+   vii. Use ellipsis (...) only to indicate a pause, as when some tool is running or being intialized
+   
+b. Abbreviations and Shortcuts
+   i. Use single-word versions when well-accepted / well-understood by users and developers. 
+   Examples, stdcell, cutline, wirelength, stdcell, flipchip, padring, bondpad, wirebond, libcell, viarule etc.
+   
+   ii. Do not abbreviate English words, expand for the sake of clarity. 
+   Incorrect: Num, # , Correct: Number
+   Incorrect: Tot., Correct: Total
+   
+   iii. Use acceptable, well-understood abbreviations for brevity. Examples, db, tech, lib, inst, term, params, etc.
+  
+   iv. Avoid contractions of action words
+   Incorrect: Can't, Don't, Can not
+   Correct: Cannot, Do not
+   
+b. Actionability
+   Messages should have clear, implied or an explicit action that is necessary for flow continuation or quality of results. 
+   Correct: A value for core_area must specified in the footprint specification, or in the environment variable CORE_AREA
+   
+d. Clarity
+   Messages must be clear and complete so as to communicate necessary and sufficient information and action. Elaborate variable, option, parameter numbers.
+   Correct: Unrecognized argument $arg, should be one of -pitch, -bump_pin_name, -spacing_to_edge, -cell_name, -bumps_per_tile, -rdl_layer, -rdl_width, -rdl_spacing
+   
+  i. Specify objects clearly in the local context
+  
+   Correct: cutWithin is smaller than cutSpacing for ADJACENTCUTS on layer {}, please check your rule definition.
+   Incomplete: Warning: {} does not have viaDef aligned with layer.
+
+   ii. Make any assumptions or use of default values explicit.
+   
+   Correct:  No net slacks found. Timing-driven mode disabled.
+   Incomplete, missing default:  Utilization exceeds 100%.
+   
+   iii. Use simple language, avoid repetitions.
+  
+   Correct: Missing orientation for cell $cell_ref
+   Incorrect: No orientation available for orientation of $cell_ref.
+   
+ 
+ Message Types
+ ~~~~~~~~~~~~~
+  
 Report
 ~~~~~~
```

Comment:
There's no space before the first char on `Message Types`. Also, both `Message Types` and `Report` have the same header level, I think `Report` should be a child of the former.